packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,831 @@
+   *
+   * @default - no KMS encryption key
+   */
+  readonly encryptionKey?: IKey;
+
+  /**
+   * The name of the global indexes set for the Global Table.
+   *
+   * Note: You must set either this property or `localIndexes` if you want permissions
+   * to be granted for indexes as well as the Global Table itself.
+   *
+   * @default - no global indexes
+   */
+  readonly globalIndexes?: string[];
+
+  /**
+   * The name of the local indexes set for the Global Table.
+   *
+   * Note: You must set either this property or `globalIndexes` if you want permissions
+   * to be granted for indexes as well as the Global Table itself.
+   *
+   * @default - no local indexes
+   */
+  readonly localIndexes?: string[]
+
+  /**
+   * Whether or not to grant permissions for all indexes of the Global Table.
+   *
+   * Note: If false, permissions will only be granted to indexes when `globalIndexes`
+   * or `localIndexes` is specified.
+   *
+   * @default false
+   */
+  readonly grantIndexPermissions?: boolean;
+}
+
+/**
+ * A Global Table.
+ */
+export class GlobalTable extends GlobalTableBase {
+  /**
+   * Creates a Global Table construct that represents an external Global Table via table name.
+   *
+   * @param scope the parent creating construct (usually `this`)
+   * @param id the construct's name
+   * @param tableName the Global Table's name
+   */
+  public static fromTableName(scope: Construct, id: string, tableName: string): IGlobalTable {
+    return GlobalTable.fromTableAttributes(scope, id, { tableName });
+  }
+
+  /**
+   * Creates a Global Table construct that represents an external Global Table via table ARN.
+   *
+   * @param scope the parent creating construct (usually `this`)
+   * @param id the construct's name
+   * @param tableArn the Global Table's ARN
+   */
+  public static fromTableArn(scope: Construct, id: string, tableArn: string): IGlobalTable {
+    return GlobalTable.fromTableAttributes(scope, id, { tableArn });
+  }
+
+  /**
+   * Creates a Global Table that represents an external Global Table.
+   *
+   * @param scope the parent creating construct (usually `this`)
+   * @param id the construct's name
+   * @param attrs attributes of the Global Table
+   */
+  public static fromTableAttributes(scope: Construct, id: string, attrs: GlobalTableAttributes): IGlobalTable {
+    class Import extends GlobalTableBase {
+      public readonly tableArn: string;
+      public readonly tableName: string;
+      public readonly tableId?: string;
+      public readonly tableStreamArn?: string;
+      public readonly encryptionKey?: IKey;
+
+      protected readonly region: string;
+      protected readonly hasIndex = (attrs.grantIndexPermissions ?? false) ||
+        (attrs.globalIndexes ?? []).length > 0 ||
+        (attrs.localIndexes ?? []).length > 0;
+
+      public constructor(tableArn: string, tableName: string, tableId?: string, tableStreamArn?: string) {
+        super(scope, id);
```

Comment:
Don't forget to pass `envFromArn` to the parent constructor.

Reply:
I'm not sure I understand. What is `envFromArn`?